rmf_traffic/src/rmf_traffic/schedule/Database.cpp

```diff
@@ -664,6 +678,36 @@ Writer::Registration Database::register_participant(
     id, state.tracker->last_known_version(), state.last_route_id);
 }
 
+
+//==============================================================================
+void Database::update_description(
+    ParticipantId id,
+    ParticipantDescription desc)
+{
+  const auto p_it = _pimpl->states.find(id);
+  if (p_it == _pimpl->states.end())
+  {
+    // *INDENT-OFF*
+    throw std::runtime_error(
+      "[Database::erase] No participant with ID ["
+      + std::to_string(id) + "]");
+    // *INDENT-ON*
+  }
+
+  const auto description_ptr =
+    std::make_shared<ParticipantDescription>(std::move(desc));
+
+  auto version = ++_pimpl->schedule_version;
+  p_it->second.last_updated = version;
+  p_it->second.description = description_ptr;
+
+  _pimpl->descriptions[id] = description_ptr;
+  _pimpl->update_participant_version.insert({id, 
```

Comment:
Using `insert` here means that nothing will change in the map if it already has an entry for `id`. I think the behavior of [`insert_or_assign`](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) is what was intended here.

I'll fix this in a commit that I'm working on right now.